lib/mixpanel-node.js

```diff
@@ -151,6 +240,128 @@ var create_client = function(token, config) {
     };
 
     /**
+     * breaks array into equal-sized chunks, with the last chunk being the remainder
+     * @param {Array} arr
+     * @param {number} size
+     * @returns {Array}
+     */
+    var chunk = function(arr, size) {
+        var chunks = [],
+            i = 0,
+            total = arr.length;
+
+        while (i < total) {
+            chunks.push(arr.slice(i, i += size));
+        }
+        return chunks;
+    };
+
+    /**
+     * helper to wait for all callbacks to complete; similar to `Promise.all`
+     * @param {Array} requests
+     * @param {Function} handler
+     * @param {Function} callback
+     */
+    var asyncAll = function(requests, handler, callback) {
+        var total = requests.length,
+            errors = null,
+            results = [],
+            done = function (err, result) {
+                if (err) {
+                    // errors are `null` unless there is an error, which allows for promisification
+                    errors = errors || [];
+                    errors.push(err);
+                }
+                results.push(result);
+                if (--total === 0) {
+                    callback(errors, results)
+                }
+            };
+
+        if (total === 0) {
+            callback(errors, results);
+        } else {
+            for(var i = 0, l = requests.length; i < l; i++) {
+                handler(requests[i], done);
+            }
+        }
+    };
+
+    /**
+     * sends events in batches
+     * @param {object}      options
+     * @param {[{}]}        options.event_list              array of event objects
+     * @param {string}      options.endpoint                e.g. `/track` or `/import`
+     * @param {number}      [options.max_async_requests]    limits concurrent async requests over the network, defaults to all async
```

Comment:
Good idea. Let's just call this `max_concurrent_requests` though. More importantly, none of the tests for `import_batch` and `track_batch` exercise this new option explicitly right now.